extensions/applicationinsights-properties-js/src/TelemetryContext.ts

```diff
@@ -62,8 +64,10 @@ export class TelemetryContext implements IPropTelemetryContext {
                 // If customer set session info, apply their context; otherwise apply context automatically generated
                 if (session && isString(session.id)) {
                     setValue(getSetValue(evt.ext, Extensions.AppExt), "sesId", session.id);
+                    this.sesId = () => session.id ;
```

Comment:
While technically this will work as applySessionContext() is currently always called after the automaticSession is updated this would be better as a first class function and have the apply use it so that everyone should be refrencing in the same way and this will (hopefully) avoid issues in the future about usages.

There will still be issues around when the automatic session is updated because the cookie expires etc, but we could address this by the below

```typescript
_self.sesId = () => {
    let session = _self.session;
    let sesId = null;
    
    // If customer set session info, apply their context; otherwise apply context automatically generated
    if (session && isString(session.id)) {
        sesId = session.id;
    } else {
        // Gets the automatic session if it exists or an empty object
        let autoSession = (_self.sessionManager || {}).automaticSession;
    
        sesId = autoSession && isString(autoSession.id) ? autoSession.id : null;
    }

    return sesId;
}

_self.applySessionContext = (evt: ITelemetryItem, itemCtx?: IProcessTelemetryContext) => {
    setValue(getSetValue(evt.ext, Extensions.AppExt), "sesId", _self.sesId(), isString);
}

Reply:
This is nice, I didn't think of this. So when the automatic session is updated, will the applySessionContext method gets called again? Is that when sesId got updated?

Reply:
Currently yes, it happens here (https://github.com/microsoft/ApplicationInsights-JS/blob/master/extensions/applicationinsights-properties-js/src/PropertiesPlugin.ts#L90-L94) where the call to _processTlemetryInternal() calls applySessionContext().

So currently it only gets updated with each event, so someone could call sesId() before an event right on the cusp of the rollover. So calling after the event will always have correct value

Reply:
I see, that makes sense. Thanks~ Updated to take the sesId out as a function.